pysat/tests/test_utils_io.py

```diff
@@ -680,75 +796,350 @@ def test_add_netcdf4_standards_to_meta(self, caplog, missing):
+                                                           'Value_Max'],
+                                             'units': ['takeout'],
+                                             'long_name': ['longer_name']}
+                                            ])
+    @pytest.mark.parametrize('assign_flag', [True, False])
+    def test_meta_translation_to_from_netcdf4(self, assign_flag, meta_trans):
+        """Test impact of meta_translation on netCDF output.
 
-        # Reset the test instrument
-        self.testInst = pysat.Instrument('pysat', 'testing2d')
-        self.testInst.load(
-            date=pysat.instruments.pysat_testing2d._test_dates[''][''],
-            use_header=True)
+        Parameters
+        ----------
+        assign_flag : bool
+            If True, assigns meta translation table as
+            ._meta_translation_table. Otherwise, uses
+            meta_translation keyword argument.
+        meta_trans : dict
+            Metadata label translation dict
 
-        # Save the un-updated metadata
-        init_meta = self.testInst.meta.copy()
+        """
 
-        # Test the initial meta data for missing Epoch time
-        assert self.testInst.index.name not in init_meta
+        # Write the file
+        pysat.utils.files.check_and_make_path(self.tempdir.name)
+        outfile = os.path.join(self.tempdir.name,
+                               'pysat_test_ncdf.nc')
 
-        # Update the metadata
-        io.add_netcdf4_standards_to_meta(self.testInst,
-                                         self.testInst.index.name)
+        if assign_flag:
+            self.testInst._meta_translation_table = meta_trans
+            pysat.utils.io.inst_to_netcdf(self.testInst, outfile)
+        else:
+            pysat.utils.io.inst_to_netcdf(self.testInst, outfile,
+                                          meta_translation=meta_trans)
 
-        # Test the metadata update
-        new_labels = ['Format', 'Var_Type', 'Time_Base', 'Time_Scale',
-                      'MonoTon', 'Depend_0', 'Depend_1', 'Display_Type']
-        assert init_meta != self.testInst.meta
-        assert self.testInst.index.name in self.testInst.meta
-        for var in init_meta.keys():
-            for label in new_labels:
-                assert label not in init_meta[var]
-                assert label in self.testInst.meta[var]
+        # Load file back and test metadata is as expected
+        with netCDF4.Dataset(outfile) as open_f:
+            for var in open_f.variables.keys():
+                test_vars = open_f[var].ncattrs()
+                # Confirm translated labels are in the file
+                # Avoid time variables
+                if 'MonoTon' not in test_vars:
+                    form = open_f[var].getncattr('Format')
+                    for key in meta_trans.keys():
+                        # String data doesn't have fill
+                        if key != 'fill' and (form != 'S1'):
```

Comment:
```suggestion
                        if key != 'fill' and form != 'S1':
```

Reply:
My experience has been that the second set needs parentheses.